hips/tiles/hipsdescription.py

```diff
@@ -0,0 +1,75 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""HiPS description class -- provides methods for parsing the HiPS file properties."""
+
+__all__ = [
+    'HipsDescription',
+]
+
+from collections import OrderedDict
+
+
+class HipsDescription:
+    """This class provides methods for parsing the HiPS properties file.
+       It also provides multiple getters around the commonly used properties.
+    """
+
+    def __init__(self, properties: str) -> None:
+        super(HipsDescription, self).__init__()
+        self.properties = self.parse_file_properties(properties)
+
+    @classmethod
+    def parse_file_properties(cls, properties: str) -> OrderedDict:
+        """Returns an ordered dictionary containing the HiPS file properties.
+
+        This method parses the HiPS file properties and loads it in a dictionary,
+        then is casts it to `collections.OrderedDict` type.
+
+        Parameters
+        ----------
+        properties : str
+            HiPS file properties
+        
+        Returns
+        -------
+        dict_properties : OrderedDict
+        """
+        properties = properties.split('\n')
+        dict_properties = {}
+        for property in properties:
+            key_value = property.split('=')
+            try:
+                dict_properties[key_value[0].strip()] = key_value[1].strip()
+            except:
+                pass
+        return OrderedDict(dict_properties)
+
+    def get_properties(self) -> OrderedDict:
```

Comment:
This isn't useful, this method can be removed.
In Python, it's considered OK / normal to just have public attributes on classes by default, and to not have any getter / setter methods. Only if needed do you "lock down" classes and try to hide the data members and use getter / setters.